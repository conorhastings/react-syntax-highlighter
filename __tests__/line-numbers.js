import React from 'react';
import renderer from 'react-test-renderer';
import SyntaxHighlighter from '../src';

const code = `const woah = fun => fun + 1;
const dude = woah(2) + 3;
function thisIsAFunction() {
  return [1,2,3].map(n => n + 1).filter(n !== 3);
}
console.log('making up fake code is really hard');

function itIs() {
  return 'no seriously really it is';
}
`;

test('SyntaxHighlighter component renders line numbers if showLineNumbers === true', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter language="javascript" showLineNumbers>
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter component does not render line numbers if showLineNumbers === undefined', () => {
  const tree = renderer
    .create(<SyntaxHighlighter language="javascript">{code}</SyntaxHighlighter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter component does not render line numbers if showInlineNumbers === true && showLineNumbers === undefined', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter language="javascript" showInlineLineNumbers={true}>
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter component renders correctly', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter language="javascript" showLineNumbers>
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter allows lineNumberProps as object', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter
        language="javascript"
        showLineNumbers={true}
        lineNumberProps={{ style: { color: 'red' } }}
      >
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter allows lineNumberProps as function', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter
        language="javascript"
        showLineNumbers
        lineNumberProps={() => ({ style: { color: 'red' } })}
      >
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('SyntaxHighlighter allows lineNumberProps as function for inline line numbers', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter
        language="javascript"
        showLineNumbers
        showInlineLineNumbers
        lineNumberProps={() => ({ style: { color: 'red' } })}
      >
        {code}
      </SyntaxHighlighter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
