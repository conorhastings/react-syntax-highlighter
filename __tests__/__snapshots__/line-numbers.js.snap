exports[`test SyntaxHighlighter component renders correctly 1`] = `
<pre
  style={
    Object {
      "color": "#444"
    }
  }>
  <code
    style={
      Object {
        "float": "left",
        "paddingRight": "10px"
      }
    }>
    <span>
      1
      
    </span>
    <span>
      2
      
    </span>
    <span>
      3
      
    </span>
    <span>
      4
      
    </span>
    <span>
      5
      
    </span>
    <span>
      6
      
    </span>
    <span>
      7
      
    </span>
    <span>
      8
      
    </span>
    <span>
      9
      
    </span>
    <span>
      10
      
    </span>
  </code>
  <code>
    <span
      style={
        Object {
          "fontWeight": "bold"
        }
      }>
      const
    </span>
     woah = fun => fun + 
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      1
    </span>
    ;
    
    <span
      style={
        Object {
          "fontWeight": "bold"
        }
      }>
      const
    </span>
     dude = woah(
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      2
    </span>
    ) + 
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      3
    </span>
    ;
    
    <span
      style={Object {}}>
      <span
        style={
          Object {
            "fontWeight": "bold"
          }
        }>
        function
      </span>
       
      <span
        style={
          Object {
            "color": "#880000",
            "fontWeight": "bold"
          }
        }>
        thisIsAFunction
      </span>
      (
      <span
        style={Object {}} />
      ) 
    </span>
    {
      
    <span
      style={
        Object {
          "fontWeight": "bold"
        }
      }>
      return
    </span>
     [
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      1
    </span>
    ,
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      2
    </span>
    ,
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      3
    </span>
    ].map(n => n + 
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      1
    </span>
    ).filter(n !== 
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      3
    </span>
    );
    }
    
    <span
      style={
        Object {
          "color": "#397300"
        }
      }>
      console
    </span>
    .log(
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      \'making up fake code is really hard\'
    </span>
    );
    
    
    <span
      style={Object {}}>
      <span
        style={
          Object {
            "fontWeight": "bold"
          }
        }>
        function
      </span>
       
      <span
        style={
          Object {
            "color": "#880000",
            "fontWeight": "bold"
          }
        }>
        itIs
      </span>
      (
      <span
        style={Object {}} />
      ) 
    </span>
    {
      
    <span
      style={
        Object {
          "fontWeight": "bold"
        }
      }>
      return
    </span>
     
    <span
      style={
        Object {
          "color": "#880000"
        }
      }>
      \'no seriously really it is\'
    </span>
    ;
    }
    
  </code>
</pre>
`;
